### PR TITLE
update_automation: make it work without git

### PR DESCRIPTION
**Pull request by bmwiedemann**
to not require SDK repo in cloud nodes

and allow to override repo and branch

### CODE REVIEW

File: scripts/jenkins/update_automation [hunk 1]
```diff
@@ -10,28 +10,41 @@
...
+: ${branch:=master}
 
 pushd ~ > /dev/null
 a_dir=${automation_repo#*git://}
 a_dir=${a_dir%.git}
 p_dir=${a_dir%/*}
 
+function do_wget
+{
+    wget -Omaster.tar.gz https://${a_dir}/archive/${branch}.tar.gz
```
Comment (aplanas) [inline scripts/jenkins/update_automation]: there is not a race here if two jobs are deployed in the same node? `mktemp`?

File: scripts/jenkins/update_automation [hunk 2]
```diff
@@ -10,28 +10,41 @@
...
+    tar xzf master.tar.gz
+    mkdir -p $p_dir
+    rm -rf $a_dir
+    mv automation-$branch $a_dir
+    exit 0
```
Comment (aplanas) [inline scripts/jenkins/update_automation]: I guess that the goal is simply end the script at this point. This makes me wonder why not to use only the github file from master as a single solution. Now there is not advantage AFACS for `git`

In any case I will suggest to `exit` outside the function, this breaks the expectations.

File: scripts/jenkins/update_automation [hunk 3]
```diff
@@ -10,29 +10,46 @@
...
+    wget -Otemp.tar.gz https://${a_dir}/archive/${branch}.tar.gz
+    tar xzf temp.tar.gz
+    rm -rf $a_dir temp.tar.gz
     mkdir -p $p_dir
-    pushd $p_dir > /dev/null
-        git clone ${automation_repo}
-    popd > /dev/null
-fi
+    mv automation-$branch $a_dir
+    exit 0
```
Comment (aplanas) [inline scripts/jenkins/update_automation]: ah very cool the `do_git || do_wget`. Now maybe we can drop this.

File: scripts/jenkins/update_automation [hunk 4]
```diff
@@ -10,29 +10,45 @@
...
+    rm -rf $a_dir ${temptar}
+    mv automation-$branch $a_dir
+}
+
+function do_git
+{
+    rpm -q git-core || zypper --non-interactive in git-core || return 24
 
+    if [ -d $a_dir ] ; then
+        pushd $a_dir > /dev/null
```
Comment (aplanas) [inline scripts/jenkins/update_automation]: maybe not needed? there is a `pushd`, `popd` wrapper in the caller, so the directory is invariant in from the caller.

Comment (jdsn) [inline scripts/jenkins/update_automation]: thats right, but does not hurt

**Pull request merged**